include/wpe/view-backend.h

```diff
@@ -97,6 +99,16 @@ WPE_EXPORT
 void
 wpe_view_backend_set_input_client(struct wpe_view_backend*, const struct wpe_view_backend_input_client*, void*);
 
+/* Sets the DOM client that listend to fullscreen requests and state changes from platform */
+WPE_EXPORT
+void
+wpe_view_backend_set_dom_fullscreen_client(struct wpe_view_backend*, const struct wpe_view_backend_fullscreen_client*, void*);
+
+/* Sets the platform function that handles fullscreen requests from DOM */
+WPE_EXPORT
+void
+wpe_view_backend_set_platform_fullscreen_handler(struct wpe_view_backend*, bool (*handler)(void*, bool), void*);
```

Comment:
Could we have a `typedef` for the callback function? I think it will be much more readable and would allow documenting it. We could add something like this, alongside the type definitions in the beginning of the file:

```c
/**
 * wpe_view_backend_fullscreen_callback:
 * @userdata: ...
 * @enable: ...
 * 
 * Type of functions used by an embedder to implement fullscreening web views.
 *
 * Returns: Whether the operation was completed.
 */
typedef bool (*wpe_view_backend_fullscreen_callback)(void *userdata, bool enable);
```

and then here, change the declaration of the function to:

```suggestion
wpe_view_backend_set_platform_fullscreen_handler(struct wpe_view_backend*,
        wpe_view_backend_fullscreen_callback callback, void* userdata);
```

Reply:
Done, thanks for the example!